File: simplified-app-shared/src/main/java/org/nypl/simplified/app/NetworkConnectivityType.java
package org.nypl.simplified.app;

/**
 * Unambiguous indication of network connectivity.
 */

public interface NetworkConnectivityType {
  /**
   * @return {@code true} iff network connectivity is available.
   */

  boolean isNetworkAvailable();

  /**
   * @return {@code true} iff WIFI network connectivity is available.
   */

  boolean isWifiAvailable();
}

File: simplified-networkconnectivity-api/src/main/java/org/nypl/simplified/networkconnectivity/api/NetworkConnectivityType.kt
package org.nypl.simplified.networkconnectivity.api

/**
 * Unambiguous indication of network connectivity.
 */

interface NetworkConnectivityType {

  /**
   * @return `true` if network connectivity is available.
   */

  val isNetworkAvailable: Boolean

  /**
   * @return `true` if WIFI network connectivity is available.
   */

  val isWifiAvailable: Boolean
}
